apps/metagpt/cli.py

```diff
@@ -128,8 +55,18 @@ def main(path_to_file: str, output_dir: str, overwrite: bool) -> None:
     try:
         # Start tool (initialise gpt client)
         gpt_updater = MetadataGPTUpdater(path_to_file)
-        # Run update
-        gpt_updater.run()
+        if gpt_updater.channel in [Channels.GARDEN, Channels.GRAPHER]:
+            # Calculate the total estimated cost
+            cost_estimated = gpt_updater.run(lazy=True)
+            # Ask the user if they want to proceed
+            proceed = input(f"The total estimated cost is ${cost_estimated:.3f}. Do you want to proceed? (yes/no): ")
+            if proceed.lower() == "yes":
+                # Actually run
+                final_cost = gpt_updater.run()
+                log.info(f"Cost GPT4: ${final_cost:.3f}")
+        else:
+            final_cost = gpt_updater.run()
+            log.info(f"Cost GPT4: ${final_cost:.3f}")
         # Save updated metadata
         gpt_updater.save_updated_metadata(output_file_path)
```

Comment:
it seems like in wizard the `gpt_updater.save_updated_metadata(output_file_path)` line never gets executed for me. I've been playing around with using the custom-written `yaml_dump` function but I don't think it actually changes anything and if I uncomment/log anything here it won't throw an error/log in the terminal. I am probably being super stupid here but I don't understand how it currently saves the updated yaml without executing this step.

Reply:
do you mean in `wizard`? Or also happening to the command client?

Reply:
yep, only in wizard! I think it's all good now though! happy to merge this one and then I'll create a new branch to play around with adapting the prompt messages